<launch>
  <arg name="rviz" default="true" />
  <arg name="base_link" value="base_link"/>
  <arg name="camera" value="pico_flexx_optical_frame"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="stereo_to_base_broadcaster" args="0.121 0.035 -0.058 -0.573 0.581 -0.414 0.403 $(arg base_link) stereo_link" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="pico_to_map_broadcaster" args="0 0 0 0.5 0.5 0.5 0.5 velodyne pico_flexx_optical_frame" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="velodyne_to_base_broadcaster" args="0 0 0 0 0 0 1 $(arg base_link) velodyne" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_map_broadcaster" args="0 0 0 0 0 0 1 map $(arg base_link)" />


  <node name="cloudTransformer" pkg="hfsd" type="cloudTransformer" output="screen">
     <param name="base_frame" value="$(arg base_link)" />
     <remap from="/cloudTransformer/inputcloud" to="/pico_flexx/points_reduced"/>
     <remap from="/velodyne_points" to ="/pico_flexx/points_reduced"/>
  </node>
  <node name="hfsd" pkg="hfsd" type="hfsd_node" output="screen">
     <param name="markerSkip" value="15" type="int"/>
     <param name="markerMag" value="1.25" type="double"/>
     <param name="markerLifetime" value="1000.0" type="double"/>
     <param name="arrowHeadDiameter" value="0.3" type="double"/>
     <param name="arrowShaftDiameter" value="0.2" type="double"/>
     <param name="arrowHeadLength" value="0.3" type="double"/>

     <param name="HREZ" value="180" type="int"/>
     <param name="VREZ" value="90" type="int"/>

     <param name="queueSize" value="10" type="int"/>

     <param name="windowSize" value="50" type="int"/>
     <param name="skipFrames" value="0" type="int"/>

     <param name="areaRestricter" value="0" type="double"/>
     <param name="heightSplitter" value="1000" type="double"/>
     <param name="minDistRejFilterVal" value="1" type="double"/>
     <param name="voxelSize" value="0.075" type="double"/>

     <param name="uniformGrid" value="false" type="bool"/>
     <param name="voxelUniformSize" value="0.075" type="double"/>

     <param name="box" value="false" type="bool"/>
     <param name="boxSizeX" value="9" type="int"/>
     <param name="boxSizeY" value="3" type="int"/>

     <param name="blur" value="false" type="bool"/>
     <param name="blurSizeX" value="3" type="int"/>
     <param name="blurSizeY" value="1" type="int"/>
     <param name="blurSigmaX" value="0" type="double"/>
     <param name="blurSigmaY" value="0" type="double"/>

     <param name="median" value="true" type="bool"/>
     <param name="medianSize" value="3" type="int"/>

     <param name="intensityOffset" value="-745.0" type="double"/>

     <param name="dilate" value="true" type="bool"/>
     <param name="dilationIterations" value="3" type="int"/>

     <param name="erode" value="false" type="bool"/>
     <param name="erosionIterations" value="1" type="int"/>

     <param name="showTiming" value="true" type="bool"/>
     <param name="showDebugMessages" value="false" type="bool"/>

     <remap from="/hfsd/points_input" to="/cloudTransformer/points"/>     
     <remap from="/hfsd/odometry_input" to="/rovio/odometry"/>
  </node>  
  
  <!--<group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args=" -d $(find hfsd)/rviz_config/hfsd_test.rviz" />
  </group>-->
  
</launch>
